miranda/units.py

```diff
@@ -79,19 +80,29 @@ def _transform(s):
 
 def get_time_frequency(d: xr.Dataset):
     freq = xr.infer_freq(d.time)
-    if freq is None:
-        raise TypeError()
+
+    # Hacky workaround for irregular Monthly data
+    if freq is None or (1 < int(freq[:-1]) < 32 and freq.endswith("D")):
```

Comment:
Are you allowed xclim? I would then use `xclim.core.calendar.parse_offset`, otherwise this is good anyway.

Reply:
Oh not his might not be good! If `freq` is anything with a base in "M", "Q" or "A", this will fail in the `int` with "Invalid literal". For example, if you do find "MS". `parse_offset` should be safer!

Reply:
If days vary between 29 and 31 between months, is `parse_offset` written to return `M`/`MS` already ? I use parse_offset just below, so if so, great!

Reply:
Ah, never mind, I think I understand. Thanks!